python/stringzilla.c

```diff
@@ -117,7 +117,9 @@ static PyTypeObject Utf8CaseInsensitiveFindIteratorType;
 static PyTypeObject HasherType;
 static PyTypeObject Sha256Type;
 
+#ifndef Py_GIL_DISABLED
 static sz_string_view_t temporary_memory = {NULL, 0};
+#endif
```

Comment:
Maybe instead we should consider thread-local variables or some other mechanism? Maybe CPython has a built-in option for each sub-interpreter?

Reply:
We could use thread-locals but they are a bit complex to cleanup again.  My strategy was to do the simple thing first.  How much does this actually help?  The malloc/free should be quite cheap.

Reply:
I've changed the PR to use thread-local storage but only when built with free-threaded Python.  This has the advantage of being fast and portable (doesn't need pthreads, for example).  But it does leak the buffer for each child thread.  If you want best performance, I think this is the way to do it.  Python has a `PyThread_tss_alloc()` function but no way to free the buffer on thread exit.